pkg/version/compatibility.go

```diff
@@ -0,0 +1,92 @@
+package version
+
+import (
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/pkg/errors"
+)
+
+type KumaDPCompatibility struct {
+	Envoy string `json:"envoy"`
+}
+
+type Compatibility struct {
+	KumaDP map[string]KumaDPCompatibility `json:"kumaDp"`
+}
+
+var CompatibilityMatrix = Compatibility{
+	KumaDP: map[string]KumaDPCompatibility{
+		"1.0.0": {
+			Envoy: "1.16.0",
+		},
+		"1.0.1": {
+			Envoy: "1.16.0",
+		},
+		"1.0.2": {
+			Envoy: "1.16.1",
+		},
+		"1.0.3": {
+			Envoy: "1.16.1",
+		},
+		"1.0.4": {
+			Envoy: "1.16.1",
+		},
+		"1.0.5": {
+			Envoy: "1.16.2",
+		},
+		"1.0.6": {
+			Envoy: "1.16.2",
+		},
+		"1.0.7": {
+			Envoy: "1.16.2",
+		},
+		"1.0.8": {
+			Envoy: "1.16.2",
+		},
+		"~1.1.0": {
+			Envoy: "~1.17.0",
+		},
+		"~1.2.0": {
+			Envoy: "~1.18.0",
+		},
+		"~1.3.0": {
+			Envoy: "~1.18.4",
+		},
+	},
+}
+
+func (c Compatibility) DP(version string) (*KumaDPCompatibility, error) {
```

Comment:
I think we can do better with the function name. This is getting the Dataplane compatibility record for the given Kuma release, right? Perhaps `DataplaneConstraints`  could work?

Reply:
Followup ... since this API isn't used currently, do you think it is likely to be used in the future?

Reply:
It might be (for example in kumactl)